src/NServiceBus.Core/Transports/Msmq/MessagePump.cs

```diff
@@ -146,18 +146,16 @@ public void Start(PushRuntimeSettings limitations)
 
                     await concurrencyLimiter.WaitAsync(cancellationToken).ConfigureAwait(false);
 
+                    var tokenSource = CancellationTokenSource.CreateLinkedTokenSource(cancellationTokenSource.Token);
+
                     var receiveTask = Task.Run(async () =>
                     {
                         try
                         {
-                            await receiveStrategy.ReceiveMessage(inputQueue, errorQueue, pipeline).ConfigureAwait(false);
+                            await receiveStrategy.ReceiveMessage(inputQueue, errorQueue, tokenSource, pipeline).ConfigureAwait(false);
```

Comment:
>I think we need to be prepared for pipelines to throw an OCE, since that would be the way to cancel the pipeline (also we need to consider that in all exception catching behaviors).

Actually since FLR catches all exceptions(yes it rethrows when giving up but SLR catches that) this seems to be a non issue since the OCE will never reach the transport no matter what happen?